packages/components/src/internal/ViewInfo.ts

```diff
@@ -98,30 +100,27 @@ export class ViewInfo {
     static BIO_DETAIL_NAME = 'BiologicsDetails';
 
     constructor(partial: Partial<ViewInfo>) {
-        // prepare name and isDefault
-        let { label, name } = partial;
-
-        if (partial.isDefault) {
-            name = ViewInfo.DEFAULT_NAME;
-            label = 'Default';
-        } else if (name === undefined || name === '') {
-            name = ViewInfo.DEFAULT_NAME;
-        } else {
-            name = partial.name;
-        }
-
-        Object.assign(this, VIEW_INFO_DEFAULTS, partial, { name, label });
+        Object.assign(this, VIEW_INFO_DEFAULTS, partial);
     }
 
     static fromJson(json: ViewInfoJson) {
-        const { columns, filter, sort, ...rest } = json;
+        const { columns, filter, label, name, sort, ...rest } = json;
         const isDefault = rest.default === true;
         delete rest.default;
+        let label_ = label;
+        let name_ = name;
+
+        if (isDefault || name === undefined || name === '') {
+            name_ = ViewInfo.DEFAULT_NAME;
+            label_ = 'Default';
```

Comment:
It was not an intentional change, but it also doesn't seem to have any effect. Just in case I reverted the logic back to the original, while also separating the logic for the name and label.